ygnmi/ygnmi_test.go

```diff
@@ -424,6 +424,38 @@ func TestLookup(t *testing.T) {
 			}
 		})
 	}
+	t.Run("use get", func(t *testing.T) {
+		fakeGNMI.Stub().GetResponse(&gpb.GetResponse{
+			Notification: []*gpb.Notification{{
+				Timestamp: 100,
+				Update: []*gpb.Update{{
+					Path: leafPath,
+					Val:  &gpb.TypedValue{Value: &gpb.TypedValue_JsonIetfVal{JsonIetfVal: []byte(`"foo"`)}},
+				}},
+			}},
+		})
+		wantGetRequest := &gpb.GetRequest{
+			Encoding: gpb.Encoding_JSON_IETF,
+			Type:     gpb.GetRequest_STATE,
+			Prefix:   &gpb.Path{},
+			Path:     []*gpb.Path{leafPath},
+		}
+		wantVal := (&ygnmi.Value[string]{
+			Path:      leafPath,
+			Timestamp: time.Unix(0, 100),
+		}).SetVal("foo")
+
+		got, err := ygnmi.Lookup(context.Background(), c, exampleocpath.Root().RemoteContainer().ALeaf().State(), ygnmi.WithUseGet())
+		if err != nil {
+			t.Fatalf("Lookup() returned unexpected error: %v", err)
+		}
+		if diff := cmp.Diff(wantVal, got, cmp.AllowUnexported(ygnmi.Value[string]{}), cmpopts.IgnoreFields(ygnmi.Value[string]{}, "RecvTimestamp"), protocmp.Transform()); diff != "" {
+			t.Errorf("Lookup() returned unexpected diff: %s", diff)
+		}
+		if diff := cmp.Diff(wantGetRequest, fakeGNMI.GetRequests()[len(fakeGNMI.GetRequests())-1], protocmp.Transform()); diff != "" {
```

Comment:
Why does it check the last request instead of the 0th one? Consider adding a helper similar to `verifySubscriptionPathsSent`, which also checks that fakeGNMI.Requests() is empty.

Reply:
changed to first